### PR TITLE
Create new EXTRA-COMMAND package

### PR DESCRIPTION
**Pull request by akohlmey**
**Summary**

As discussed at the LAMMPS Master Class workshop the group2ndx and ndx2group commands should be moved to the EXTRA-COMMAND package. No other current commands make sense to be moved there.

**Related Issue(s)**

N/A

**Author(s)**

Axel Kohlmeyer, Temple U

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

N/A

**Implementation Notes**

There are some updates to error handling. Gromacs does not allow group names with space (LAMMPS apparently does even though I have not seen anybody using that). _ndx2group_ would already throw a cryptic error message on such files, this is now down in a more consistent way with a meaningful error message. _group2ndx_ will now skip over processing groups with whitespace in the name.

**Post Submission Checklist**

- [x] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [x] The source code follows the LAMMPS formatting guidelines
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [x] The added/updated documentation is integrated and tested with the documentation build system
- [x] The feature has been verified to work with the conventional build system
- [x] The feature has been verified to work with the CMake based build system
- [ ] Suitable tests have been added to the unittest tree.
- [ ] A package specific README file has been included or updated
- [ ] One or more example input decks are included

### CODE REVIEW

File: doc/src/Packages_details.rst [hunk 1]
```diff
@@ -886,6 +887,22 @@ This package has :ref:`specific installation instructions <electrode>` on the
...
 ----------
 
+.. _PKG-EXTRA-COMMAND:
+
+EXTRA-COMMAND package
+---------------------
+
+**Contents:**
+
+Additional compute styles that are less commonly used.
```
Comment (giacomofiorin) [inline doc/src/Packages_details.rst]: Did you mean specifically "compute styles" instead of "commands"?

**Review by giacomofiorin — APPROVED** [review]
As discussed earlier, these two commands are much more broadly useful than just supporting Colvars.

Thank you for moving them!

File: doc/src/Packages_details.rst (hunk 1 above)
Comment (akohlmey) [reply doc/src/Packages_details.rst]: No. This is a copy-n-paste error.

**Comment by sjplimp** [comment]
@akohlmey @giacomofiorin I think the doc page for the 2 commands should be expanded in 3 ways:
a) Give some sample use cases for these commands: for Gromacs compatibility or data exchanges, for Colvars, and for other use cases you envision
b) Describe the format of the file - if you don't do this then it is only a Gromacs-related tool and someone has to be familiar with Gromacs to use the command
c) Describe how to output this file from Gromacs, or to use it with Gromacs if LAMMPS outputs an altered file

**Comment by giacomofiorin** [comment]
> @akohlmey @giacomofiorin I think the doc page for the 2 commands should be expanded in 3 ways: a) Give some sample use cases for these commands: for Gromacs compatibility or data exchanges, for Colvars, and for other use cases you envision b) Describe the format of the file - if you don't do this then it is only a Gromacs-related tool and someone has to be familiar with Gromacs to use the command c) Describe how to output this file from Gromacs, or to use it with Gromacs if LAMMPS outputs an altered file

@sjplimp Good suggestions.

@akohlmey I don't mind taking a stab at this, since we have already content from the Colvars doc. Let me know what you think.

**Comment by akohlmey** [comment]
> Let me know what you think.

@giacomofiorin please go ahead. I have a few ideas of my own, but I can always edit them in later.

**Comment by akohlmey** [comment]
@giacomofiorin didn't have the patience to wait any longer and added the changes I have in mind. Please let me know if you want to make any changes/updates to this.

**Comment by giacomofiorin** [comment]
> @giacomofiorin didn't have the patience to wait any longer 

:-D  Okay, I guess.

> and added the changes I have in mind. Please let me know if you want to make any changes/updates to this.

What you wrote looks more or less like part of what I wanted, too.

The main reason why I haven't pushed anything yet (other than trying to finish unrelated work first) is that I also wanted to confirm how these two LAMMPS commands handle spaces within a group's name.

Because the GROMACS `make_ndx` tool does not allow them, they are not checking for them, which can lead to some interesting errors: 
https://gromacs.bioexcel.eu/t/colvar-output-force-and-work/3191/10

**Comment by giacomofiorin** [comment]
LGTM

**Comment by akohlmey** [comment]
> The main reason why I haven't pushed anything yet (other than trying to finish unrelated work first) is that I also wanted to confirm how these two LAMMPS commands handle spaces within a group's name.

That is an interesting subject.
I have never seen group names with spaces, but apparently that is allowed in LAMMPS if you put the name in quotes (I just checked it).

The ndx2group command would throw an error if there was a section with two or more words, but a cryptic one, so I changed the code to have more meaningful error messages.

The group2ndx command would happily create an index file with an invalid header, so I changed the code to skip over groups with whitespace in their name. The alternative would be to replace with whitespace with a non-whitespace character, e.g. '_' an underscore. But I'll think about this in case someone is actually complaining about not being able to write out groups with spaces in them.

**Comment by akohlmey** [comment]
@sjplimp does this address your concerns?

**Pull request merged**